### PR TITLE
Remove CLS and use asyncHooks

### PR DESCRIPTION
**Pull request by cdupuis**

### CODE REVIEW

File: src/internal/util/cls.ts [hunk 1]
```diff
@@ -1,27 +1,17 @@
-import {
-    createNamespace,
-    getNamespace,
-    Namespace,
-} from "continuation-local-storage";
+import * as asyncHooks from "async_hooks";
 
-// create the local storage namespace
-createNamespace("automation-client");
+const namespaces = {};
```
Comment (ddgenome) [inline src/internal/util/cls.ts]: Is this of type `Namespace`?

File: src/internal/util/cls.ts [hunk 2]
```diff
@@ -1,27 +1,17 @@
...
 
-export function init(): Namespace {
-    return getNamespace("automation-client");
+export function create() {
```
Comment (ddgenome) [inline src/internal/util/cls.ts]: Should this return `: Namespace`?

File: src/internal/util/cls.ts [hunk 3]
```diff
@@ -36,3 +26,62 @@ export interface AutomationContext {
     ts: number;
 
 }
+
+class Namespace {
+
+    constructor(private readonly context = {}) {
+    }
+
+    public run(fn) {
```
Comment (ddgenome) [inline src/internal/util/cls.ts]: Is there a reason none of these functions have their return type specified.

**Comment by cdupuis** [comment]
Sorry, this code is highly experimental. Didn’t intent to raise a PR yet. Needs a lot more testing to see if it actually works.

**Pull request closed without merging**